src/mito2/src/read/seq_scan.rs

```diff
@@ -187,52 +220,159 @@ impl SeqScan {
+                    metrics.num_batches += 1;
+                    metrics.num_rows += batch.num_rows();
+
+                    let convert_start = Instant::now();
+                    let record_batch = stream_ctx.input.mapper.convert(&batch, cache)?;
+                    metrics.convert_cost += convert_start.elapsed();
+                    yield record_batch;
+
+                    fetch_start = Instant::now();
+                }
+                metrics.scan_cost += fetch_start.elapsed();
+                metrics.total_cost = stream_ctx.query_start.elapsed();
+                metrics.observe_metrics_on_finish();
+
+                debug!(
+                    "Seq scan finished, region_id: {:?}, partition: {}, metrics: {:?}",
+                    stream_ctx.input.mapper.metadata().region_id,
+                    partition,
+                    metrics,
+                );
+            }
+        };
+
+        let stream = Box::pin(RecordBatchStreamWrapper::new(
+            self.stream_ctx.input.mapper.output_schema(),
+            Box::pin(stream),
+        ));
+
+        Ok(stream)
+    }
+
+    /// Scans the given partition when the part list is not set.
+    /// This method will do a lazy initialize of part list and
+    /// ignores the partition settings in `properties`.
+    fn uncached_scan_part_impl(
+        &self,
+        partition: usize,
+    ) -> Result<SendableRecordBatchStream, BoxedError> {
+        let num_partitions = self.properties.ranges.len();
+        if partition >= num_partitions {
+            return Err(BoxedError::new(
+                PartitionOutOfRangeSnafu {
+                    given: partition,
+                    all: self.properties.ranges.len(),
+                }
+                .build(),
+            ));
+        }
+        let mut metrics = ScannerMetrics {
+            prepare_scan_cost: self.stream_ctx.prepare_scan_cost,
+            ..Default::default()
+        };
+        let stream_ctx = self.stream_ctx.clone();
+        let semaphore = self.semaphore.clone();
+
+        // build stream
+        let stream = try_stream! {
+            // init parts
+            let parts_len = {
+                let mut parts = stream_ctx.parts.lock().await;
+                Self::maybe_init_parts(&stream_ctx.input, &mut parts, &mut metrics).await
+                    .map_err(BoxedError::new)
+                    .context(ExternalSnafu)?;
+                parts.len()
             };
-            let cache = stream_ctx.input.cache_manager.as_deref();
-            let mut fetch_start = Instant::now();
-            while let Some(batch) = reader
-                .next_batch()
+
+            for id in (0..parts_len).skip(partition).step_by(num_partitions) {
+                let maybe_reader = Self::build_merge_reader(
+                    &stream_ctx,
+                    id,
+                    semaphore.clone(),
+                    &mut metrics,
+                )
```

Comment:
We need to use the merge reader to merge all `ScanPart` under the same `partition`.